packages/web3-core-method/src/index.js

```diff
@@ -889,47 +845,39 @@ function _handleTxPricing(method, tx) {
                 params: 0
             })).createFunction(method.requestManager);
 
-            if (tx.type < '0x2' && tx.gasPrice !== undefined) {
-                // Legacy transaction, return provided gasPrice
-                resolve({ gasPrice: tx.gasPrice })
-            } else {
-                Promise.all([
-                    getBlockByNumber(),
-                    getGasPrice()
-                ]).then(responses => {
-                    const [block, gasPrice] = responses;
-                    if (
-                        (tx.type === '0x2') &&
-                        block && block.baseFeePerGas
-                    ) {
-                        // The network supports EIP-1559
-    
-                        // Taken from https://github.com/ethers-io/ethers.js/blob/ba6854bdd5a912fe873d5da494cb5c62c190adde/packages/abstract-provider/src.ts/index.ts#L230
-                        let maxPriorityFeePerGas, maxFeePerGas;
-    
-                        if (tx.gasPrice) {
-                            // Using legacy gasPrice property on an eip-1559 network,
-                            // so use gasPrice as both fee properties
-                            maxPriorityFeePerGas = tx.gasPrice;
-                            maxFeePerGas = tx.gasPrice;
-                            delete tx.gasPrice;
-                        } else {
-                            maxPriorityFeePerGas = tx.maxPriorityFeePerGas || '0x3B9ACA00'; // 1 Gwei
-                            maxFeePerGas = tx.maxFeePerGas ||
-                                utils.toHex(
-                                    utils.toBN(block.baseFeePerGas)
-                                        .mul(utils.toBN(2))
-                                        .add(utils.toBN(maxPriorityFeePerGas))
-                                );
-                        }
-                        resolve({ maxFeePerGas, maxPriorityFeePerGas });
+            Promise.all([
+                getBlockByNumber(),
+                getGasPrice()
+            ]).then(responses => {
+                const [block, gasPrice] = responses;
+                if (block && block.baseFeePerGas) {
+                    // The network supports EIP-1559
+
+                    // Taken from https://github.com/ethers-io/ethers.js/blob/ba6854bdd5a912fe873d5da494cb5c62c190adde/packages/abstract-provider/src.ts/index.ts#L230
+                    let maxPriorityFeePerGas, maxFeePerGas;
+
+                    if (tx.gasPrice) {
```

Comment:
Maybe we should check for eip1559 support and then default to that, else no support offer gasPrice?

Reply:
This part of the code represents a user passing `gasPrice` on a network that supports EIP-1559, so we set the EIP-1559 related properties (`maxPriorityFeePerGas` and `maxFeePerGas`) to what the user gave us as `gasPrice`

We check for EIP-1559 support [here](https://github.com/ChainSafe/web3.js/blob/ce2de6dcbc8ebfa0228ebaaef25ee0acb62be598/packages/web3-core-method/src/index.js#L853) and if it's not supported we return `gasPrice` [here](https://github.com/ChainSafe/web3.js/blob/ce2de6dcbc8ebfa0228ebaaef25ee0acb62be598/packages/web3-core-method/src/index.js#L878)

Reply:
Yep you're right